src/embezzler.ts

```diff
@@ -340,3 +353,57 @@ export function estimatedTurns(): number {
 
   return turns;
 }
+
+/**
+ * Gets next available embezzler fight. If there is no way to generate a fight, but copies are available,
+ * the user is prompted to purchase a pocket wish to start the embezzler chain.
+ * @returns the next available embezzler fight
+ */
+export function getNextEmbezzlerFight(): EmbezzlerFight | null {
+  for (const fight of embezzlerSources) {
+    if (fight.available()) return fight;
+  }
+  const potential = embezzlerCount();
+  const averageEmbezzlerNet = ((baseMeat + 750) * meatDropModifier()) / 100;
+  if (potential > 0) {
+    print(`You have the following embezzler-sources untapped right now:`, "blue");
+    embezzlerSources
+      .filter((source) => !source.available() && source.potential() > 0)
+      .map((source) => `${source.potential()} from ${source.name}`)
+      .forEach((text) => print(text, "blue"));
+    if (
+      get("_genieFightsUsed") < 3 &&
+      userConfirm(
+        `Garbo has detected you have ${potential} potential ways to copy an Embezzler, but no way to start a fight with one. Current embezzler net (before potions) is ${averageEmbezzlerNet}. Should we wish for an Embezzler?`
+      )
+    ) {
+      acquire(1, $item`pocket wish`, WISH_VALUE);
+      const id = "Pocket Wish";
+      const wishFight = new EmbezzlerFight(
+        id,
+        () => true, // available so that future getNextEmbezzlerFight requests don't queue up more pocket wish KGEs
+        () => 1,
+        () => {
+          visitUrl(`inv_use.php?pwd=${myHash()}&which=3&whichitem=9537`, false, true);
+          visitUrl(
+            "choice.php?pwd&whichchoice=1267&option=1&wish=to fight a Knob Goblin Embezzler ",
+            true,
+            true
+          );
+          visitUrl("main.php", false);
+          runCombat();
+
+          // remove the pocket wish embezzler source
+          const index = embezzlerSources.findIndex((source) => source.name === id);
+          embezzlerSources.splice(index, 1);
+        }
+      );
+
+      // add the pocket wish as the first embezzler source
+      embezzlerSources.splice(0, 0, wishFight);
```

Comment:
I am not sure why we are adding & removing the wishFight from embezzlerSources. Is just returning the fight insufficient?

Reply:
The problem with just returning the fight is that there are cases where the user will get prompted multiple times. Any time we come to check the list of sources, the user would get prompted again. If you prefer to use a property or something or session variable, I'm fine doing that.

Reply:
I think a session variable is our best bet. You could make it part of globalOptions, or you could not make it part of globalOptions. Lots of options here!

My worry is that adding and removing it from the array is, well, a little arcane.